application/backup.sh

```diff
@@ -1,114 +1,133 @@
 #!/usr/bin/env bash
 
-# Send error to Sentry
+# Initialize logging with timestamp
+log() {
+    local level="${1:-INFO}"
+    local message="$2"
+    echo "[$(date '+%Y-%m-%d %H:%M:%S')] ${level}: ${message}"
+}
+
+# Sentry reporting with validation and backwards compatibility
 error_to_sentry() {
     local error_message="$1"
     local db_name="$2"
     local status_code="$3"
 
-if [ ! -z "${SENTRY_DSN}" ]; then
-    wget -q --header="Content-Type: application/json" \
-         --post-data="{
-            \"message\": \"${error_message}\",
-            \"level\": \"error\",
-            \"extra\": {
-                \"database\": \"${db_name}\",
-                \"status_code\": \"${status_code}\",
-                \"hostname\": \"${HOSTNAME}\"
-                }
-}" \
-         -O - "${SENTRY_DSN}"
-fi
+    # Check if SENTRY_DSN is configured - ensures backup continues
+    if [ -z "${SENTRY_DSN:-}" ]; then
+        log "DEBUG" "Sentry logging skipped - SENTRY_DSN not configured"
+        return 0
+    fi
+
+    # Validate SENTRY_DSN format
+    if ! [[ "${SENTRY_DSN}" =~ ^https://[^@]+@[^/]+/[0-9]+$ ]]; then
+        log "WARN" "Invalid SENTRY_DSN format - Sentry logging will be skipped"
+        return 0
+    fi
+
+    # Attempt to send event to Sentry
+    if sentry-cli send-event \
+        --message="${error_message}" \
+        --level=error \
```

Comment:
This seems to work as it is, but for consistency with other options and with the documentation, let's remove the `=`. BTW, good idea using the long form of the option.
```suggestion
        --message "${error_message}" \
        --level error \
```